src/app/code/community/Omise/Gateway/Controller/Base.php

```diff
@@ -109,9 +109,12 @@ protected function checkPaymentStatusAndUpdateOrder($charge, $order) {
      * If payment is awaiting to capture then updating order status as 'payment_review'. In case of 3-D secured payment,
      * it will be 'processing'.
      * @param Omise_Gateway_Model_Order $order
+     * @param Omise_Gateway_Model_Api_Charge $charge
      * @return Mage_Core_Controller_Varien_Action
      */
-    protected function paymentAwaiting($order) {
+    protected function paymentAwaiting($order, $charge) {
+        if($charge->isUnauthorized() && $order->getStatus() != Mage_Sales_Model_Order::STATE_CANCELED)
+            return $this->cancelOrder($order, $charge);
```

Comment:
Charge unauthorised does not always mean "cancel". It can be a case of "waiting for the bank issuer to process the payment." Cancelling an order right away sounds incorrect behaviour.